nixos/modules/services/hardware/logiops.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) mkEnableOption mkIf mkOption types;
+  cfg = config.services.logiops;
+  format = pkgs.formats.libconfig { };
+in {
+  options.services.logiops = {
+    enable = mkEnableOption
+      "logiops, an unofficial userspace driver for HID++ Logitech devices";
+
+    settings = mkOption {
+      type = types.submodule { freeformType = format.type; };
+      description = ''
+        Configuration for logiops.
+
+        See <https://github.com/PixlOne/logiops/wiki/Configuration> for more details.
+        Also see <https://github.com/PixlOne/logiops/blob/main/logid.example.cfg> for an example config.
+      '';
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc."logid.cfg".source =
+      format.generate "logid.cfg" cfg.settings;
+    systemd.services.logiops = {
+      description = "Logitech Configuration Daemon";
+      documentation = [ "https://github.com/PixlOne/logiops" ];
+
+      wantedBy = [ "graphical.target" ];
+
+      startLimitIntervalSec = 0;
+      after = [ "multi-user.target" ];
+      wants = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.logiops}/bin/logid";
+        Restart = "always";
+        User = "root";
+
+        RuntimeDirectory = "logiops";
+
+        CapabilityBoundingSet = [ "CAP_SYS_NICE" ];
+        DeviceAllow = [ "/dev/uinput rw" "char-hidraw rw" ];
+        ProtectClock = true;
+        PrivateNetwork = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        PrivateUsers = true;
+        PrivateMounts = true;
+        PrivateTmp = true;
+        RestrictNamespaces = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectControlGroups = true;
+        MemoryDenyWriteExecute = true;
+        RestrictRealtime = true;
+        LockPersonality = true;
+        ProtectProc = "invisible";
+        SystemCallFilter = [ "nice" "@system-service" "~@privileged" ];
+        RestrictAddressFamilies = [ "AF_NETLINK" "AF_UNIX" ];
+        RestrictSUIDSGID = true;
+        NoNewPrivileges = true;
+        ProtectSystem = "strict";
+        ProcSubset = "pid";
+        UMask = "0077";
+      };
+    };
+
+    # Add a `udev` rule to restart `logiops` when the mouse is connected
+    # https://github.com/PixlOne/logiops/issues/239#issuecomment-1044122412
+    services.udev.extraRules = ''
+      ACTION=="add", SUBSYSTEM=="input", ATTRS{id/vendor}=="046d", RUN{program}="${pkgs.systemd}/bin/systemctl --no-block restart logiops.service"
+    '';
```

Comment:
Root comment stuff aside, why v0.3.3? It's deeply broken (& project hasn't been touched in 7 months), [I've bisected it before](https://github.com/PixlOne/logiops/issues/409). And you mean `try-restart`.

Reply:
It just happened to be the latest version and I didn't look into it too much; I tried `v0.3.3` and it seemed to work for myself with this `udev` rule in place.